src/main/webapp/script.js

```diff
@@ -217,54 +217,105 @@ function getGraphDisplay(graphNodes, graphEdges) {
   // When the user clicks on a node, display the token list tooltip for the node
   cy.on('tap', 'node', function (evt) {
     const node = evt.target;
+    console.log(node);
     node.tip.show();
   });
 
-  const searchElement = document.getElementById('search');
-  document.getElementById('search-button').onclick = function() {
-    if (searchNode(cy, searchElement.value) || searchElement.value == "") {
-      document.getElementById('search-error').innerText = "";
-    } else {
-      document.getElementById('search-error').innerText = "Node does not exist.";
-    }
-  };
-  
+  document.getElementById('reset').onclick = function(){ resetElements(cy) };
+
+  document.getElementById('search-button').onclick = function() { search(cy, "node", searchNode) };
+
+  document.getElementById('search-token-button').onclick = function() { search(cy, "token", searchToken) };
 }
 
 /**
- * Zooms in on specific node
+ * Searches based on type (node or token)
  */
-function searchNode(cy, query) {
-  // reset nodes to default color
-  cy.nodes().forEach(node => {
-    node.style('background-color', 'blue');
-    node.style('opacity', '1')
-  });
-  const target = findNodeInGraph(cy, query);
-  if (target) {
-    cy.nodes().forEach(node => node.style('opacity', '0.25'));
-    target.style('background-color', 'olive');
-    target.style('opacity', '1');
-    cy.fit(target, 50);
-    return true;
+function search(cy, type, searchFunction) {
+  resetElements(cy);
+  let errorText;
+  let query = document.getElementById(type + '-search').value;
+  let result = searchFunction(cy, query);
+  if (query == "" || result) {
```

Comment:
Since searching for the empty string is not allowed, would it be better to put the `if (query == "")` test before we call the search function to save some work?